src/platform/unix/mod.rs

```diff
@@ -471,8 +471,18 @@ impl OsIpcReceiverSet {
     pub fn select(&mut self) -> Result<Vec<OsIpcSelectionResult>,UnixError> {
         let mut selection_results = Vec::new();
         match self.poll.poll(&mut self.events, None) {
-            Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
+            Ok(sz) => {
+                if sz < 1 {
+                    return self.select();
+                }
+            },
+            Err(ref e) => {
+                if e.kind() == ErrorKind::Interrupted {
+                    return self.select();
```

Comment:
Indeed this is not even a real tail call (as @mbrubeck pointed out on IRC), since the `Vec` has a destructor -- so even in optimised builds this clutters the stack... Not much of an issue here, as it's a rare exceptional situation, so the worst thing I guess is the bloated backtrace -- but it's a good thing to keep in mind in general I guess :-)